File: features/advisor/src/main/java/com/nivasafinance/features/advisor/repository/AdvisorRepository.java
package com.nivasafinance.features.advisor.repository;

import com.nivasafinance.features.advisor.entity.Advisor;
import org.javers.spring.annotation.JaversSpringDataAuditable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
@JaversSpringDataAuditable
public interface AdvisorRepository extends JpaRepository<Advisor, UUID> {
    @Query("SELECT a FROM Advisor a WHERE a.identifier = :identifier AND a.isDeleted = false")
    Optional<Advisor> findByIdentifier(@Param("identifier") UUID identifier);

    @Query("SELECT a FROM Advisor a WHERE a.username = :username AND a.isDeleted = false")
    Optional<Advisor> findByUsername(@Param("username") String username);

    // Unfiltered — used by admin delete/undo-delete and creation uniqueness checks
    @Query("SELECT a FROM Advisor a WHERE a.identifier = :identifier")
    Optional<Advisor> findByIdentifierIncludingDeleted(@Param("identifier") UUID identifier);

    @Query("SELECT a FROM Advisor a WHERE a.username = :username")
    Optional<Advisor> findByUsernameIncludingDeleted(@Param("username") String username);
}

File: features/advisor/src/main/java/com/nivasafinance/features/advisor/repository/AdvisorRepository.kt
package com.nivasafinance.features.advisor.repository

import com.nivasafinance.features.advisor.entity.Advisor
import org.javers.spring.annotation.JaversSpringDataAuditable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
@JaversSpringDataAuditable
interface AdvisorRepository : JpaRepository<Advisor, UUID> {

    @Query(
        "SELECT a.* FROM advisor a JOIN person p ON a.person_id = p.id " +
            "WHERE p.mobile_numbers @> jsonb_build_array(" +
            "jsonb_build_object('isPrimary', true, 'number', :primaryMobileNo))",
        nativeQuery = true
    )
    fun findByPrimaryMobileNo(primaryMobileNo: String): Advisor?

    fun findByPersonId(personId: UUID): Advisor?
}
